Tests/CursorTests.swift

```diff
@@ -0,0 +1,194 @@
+import XCTest
+import PusherPlatform
+@testable import PusherChatkit
+
+var alice: PCCurrentUser?
+var bob: PCCurrentUser?
+var roomId: Int?
+
+class AliceRoomDelegate: NSObject, PCRoomDelegate {
+    let ex: XCTestExpectation?
+
+    init(expectation: XCTestExpectation? = nil) {
+        ex = expectation
+    }
+
+    func newCursor(cursor: PCCursor) {
+        XCTAssertEqual(cursor.position, 42)
+        if let e = ex {
+            e.fulfill()
+        }
+    }
+}
+
+class CursorTests: XCTestCase {
+    override func setUp() {
+        super.setUp()
+
+        let deleteResourcesEx = expectation(description: "delete resources")
+        let createRolesEx = expectation(description: "create roles")
+        let createAliceEx = expectation(description: "create Alice")
+        let createBobEx = expectation(description: "create Bob")
+
+        deleteInstanceResources() { err in
+            guard err == nil else {
+                fatalError(err!.localizedDescription)
+            }
+            deleteResourcesEx.fulfill()
+
+            createStandardInstanceRoles() { err in
+                guard err == nil else {
+                    fatalError(err!.localizedDescription)
+                }
+                createRolesEx.fulfill()
+            }
+
+            createUser(id: "alice") { err in
+                guard err == nil else {
+                    fatalError(err!.localizedDescription)
+                }
+                createAliceEx.fulfill()
+            }
+
+            createUser(id: "bob") { err in
+                guard err == nil else {
+                    fatalError(err!.localizedDescription)
+                }
+                createBobEx.fulfill()
+            }
+        }
+
+        waitForExpectations(timeout: 10)
+
+        alice = user(id: "alice")
+        bob = user(id: "bob")
+        roomId = createRoom(user: user(id: "alice"), roomName: "mushroom", addUserIds: ["bob"]).id
+    }
+
+    func testOwnReadCursorUndefinedIfNotSet() {
+        let cursor = try! alice?.readCursor(roomId: roomId!)
+        XCTAssertNil(cursor)
+    }
+
+    // TODO hook for setting own read cursor? (currently unsupported by the looks of it)
+
+    func testGetOwnReadCursor() {
+        let ex = expectation(description: "got own read cursor")
+
+        alice?.setReadCursor(position: 42, roomId: roomId!) { error in
+            XCTAssertNil(error)
+
+            sleep(1) // give the read cursor a chance to propagate down the connection
```

Comment:
Is this required if we're in the completion handler closure? Presumably it should already be set and so we don't need to sleep?

Reply:
If you remove it the tests fail some of the time, I think that what's happening is the request has succeeded, but we haven't received the cursor back on the subscription yet, and _thats_ when it gets updated internally. In fact we would always need at least a small timeout because if you're setting a cursor as one user and reading it as another there's always going to be a small gap between the request suceeding and the state updating.

Reply:
@hamchapman this has been marked out of date because of an unrelated change, pinging you so it doesn't get lost

Reply:
Ah good point - guess we'll have to live with it for now then